File: fake-store-quarkus/src/main/java/org/arielzarate/interfaces/rest/dto/ProductRequest.java
package org.arielzarate.interfaces.rest.dto;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductRequest {
    private String title;
    private Double price;
    private String description;
    private String category;
    private String image;
    private RatingRequest rating;

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class RatingRequest {
        private Double rate;
        private Integer count;
    }
}

File: api-fake-hexagonal/src/main/kotlin/com/arielZarate/hexagonal_example/interfaces/model/ProductRequest.kt
package com.arielZarate.hexagonal_example.interfaces.model


data class ProductRequest(
    val title: String,
    val price: Double,
    val description: String,
    val category: String,
    val image: String,
    val rating: RatingRequest
) {
    data class RatingRequest(val rate: Double, val count: Int)
}